packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaForm.tsx

```diff
@@ -68,53 +83,86 @@ export type SubmitHandle = {
 type TableColumnSchemaFormProps = {
   /* The type of the Table, which determines various schema restrictions and form functionality */
   entityType: EntityType
+  /* If this is an entity view, the ViewScope can be used to determine the default column models and fetch annotation column models */
+  viewScope?: ViewScope
   initialData?: ColumnModel[]
   onSubmit: (formData: ColumnModelFormData[]) => void
+  isSubmitting: boolean
 }
 
+const ColumnHeader: StyledComponent<BoxProps> = styled(Box, {
+  label: 'ColumnHeader',
+})({
+  fontWeight: 700,
+})
+
 const TableColumnSchemaForm = React.forwardRef<
   SubmitHandle,
   TableColumnSchemaFormProps
 >(function TableColumnSchemaForm(props, ref) {
-  const { initialData, entityType, onSubmit } = props
+  const { initialData, entityType, viewScope, onSubmit, isSubmitting } = props
 
   const numColumnModels = useAtomValue(
     useMemo(() => atom(get => get(tableColumnSchemaFormDataAtom).length), []),
   )
 
   const dispatch = useSetAtom(tableColumnSchemaFormDataAtom)
 
+  // useAtomCallback will let us imperatively read the form data, instead of tracking it in state and triggering a full re-render of the form when any data changes
+  const readFormData = useAtomCallback(
+    useCallback(get => {
+      return get(tableColumnSchemaFormDataAtom)
+    }, []),
+  )
+  const concreteTableType = convertToConcreteEntityType(entityType)
+  const isView = VIEW_CONCRETE_TYPE_VALUES.includes(
+    concreteTableType as (typeof VIEW_CONCRETE_TYPE_VALUES)[number],
+  )
+  const isSubmissionView =
+    SUBMISSION_VIEW_CONCRETE_TYPE_VALUE == concreteTableType
+  const hasDefaultColumnModels = isView
+  const { data: defaultColumnModels, isLoading: isLoadingDefaultColumns } =
+    useGetDefaultColumnModels(
+      viewScope?.viewEntityType!,
+      viewScope?.viewTypeMask,
+      {
+        enabled: hasDefaultColumnModels,
+        staleTime: Infinity, // The default column models will never change
+      },
+    )
+
+  const hasAnnotationColumnModels = isView && !isSubmissionView
+  const {
+    data: annotationColumnModels,
```

Comment:
Annotation data is present so we can easily validate against sizes when we tackle SWC-6602

Reply:
Nice prep, thank you